assets/js/atomic/blocks/product-elements/rating/edit.tsx

```diff
@@ -65,11 +84,26 @@ const Edit = ( {
 		</>
 	);
 };
-export default withProductSelector( {
-	icon: BLOCK_ICON,
-	label: BLOCK_TITLE,
-	description: __(
-		'Choose a product to display its rating.',
-		'woo-gutenberg-products-block'
-	),
-} )( Edit );
+
+/**
+ * The Edit block can be used in several contexts.
+ *
+ * If the `isDescendentOfSingleProductTemplate` attribute is true, it returns the `Edit` component with the `props` passed in.
+ * If it's false, it wraps the `Edit` component with the `withProductSelector` higher-order component, which provides a product selector UI for the user to choose a product to display its rating.
+ *
+ */
+export default (
+	props: BlockEditProps< BlockAttributes > & { context: Context }
+) => {
+	if ( props.attributes.isDescendentOfSingleProductTemplate ) {
+		return <Edit { ...props } />;
+	}
+	return withProductSelector( {
+		icon: BLOCK_ICON,
+		label: BLOCK_TITLE,
+		description: __(
+			'Choose a product to display its rating.',
+			'woo-gutenberg-products-block'
+		),
+	} )( Edit )( props );
+};
```

Comment:
This is not necessary if we set [the ancestors](https://github.com/woocommerce/woocommerce-blocks/pull/9499/files#diff-a1124248b7c49afbad741e61af47756aa717955fe8180c77a825b01da9fbaf06R18-R23).

Reply:
Just explaining why I had to to this: Currently, we are using the `withProductSelector` HOC around the Edit component. This is being done because when the block is added to a component that has Product Data context, it will not require the user to select a product. However, since we are now using the block inside Single Product Template, it doesn't have the product context (because the product can be anything) so, by default, the Product Selector will appear.

If you look [here at this code](https://github.com/woocommerce/woocommerce-blocks/pull/9499/files/5ec7b22a8d1977bd7790f27b816f1b06f5fa084d#diff-4e154173c83a90857629f0bdf60e0af8dbd33560750747056a1535ad9132e966R95-R109) you'll see that I had to put a conditional that if the `isDescendentOfSingleProductTemplate` attribute is true, the block should render without using the withProductSelector HOC. I had to do this because when the HOC is used, the Edit component will only render if the user selects a product from the Product selector, but we really need the Edit component to render since the logic to set the `isDescendentOfSingleProductTemplate` is inside it.

Considering your comment about the `ancestor` key being used, I'll be removing this logic then, since it will only be used inside blocks that have product data context.